### PR TITLE
#134 MySql table charset

### LINKED ISSUE #134: Enable to specify charset for MySqlTables
Rationale: VSMYSQL uses the constructor of `MySqlSchema` to inject a writer using a specific charset to enable tests using a specific charset. [MySQLSqlDialectIT](https://github.com/exasol/mysql-virtual-schema/blob/main/src/test/java/com/exasol/adapter/dialects/mysql/MySQLSqlDialectIT.java#L222) is currently broken as the constructor has been made private in release 3.5.2 of TDBJ.

Potential classes: MySqlObjectFactory or MySqlSchema.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #134

### CODE REVIEW

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import com.exasol.db.Identifier;
+import com.exasol.dbbuilder.dialects.*;
+
+/**
+ * A MySql table that allows specifying a charset.
+ */
+public class MySqlTable extends Table {
+
+    private final String charset;
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:11]:
```suggestion
public class MySqlTable extends Table {
    private final String charset;
```

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package com.exasol.dbbuilder.dialects.mysql;
+
+import com.exasol.db.Identifier;
+import com.exasol.dbbuilder.dialects.*;
+
+/**
+ * A MySql table that allows specifying a charset.
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:7]:
```suggestion
 * A MySql table that allows specifying a character set.
```

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ */
+public class MySqlTable extends Table {
+
+    private final String charset;
+
+    /**
+     * Create a new MySql table based on a given builder.
+     *
+     * @param builder the builder from which to copy the values
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:16]:
```suggestion
     * @param builder builder from which to copy the values
```

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    protected MySqlTable(final Builder builder) {
+        super(builder);
+        this.charset = builder.charset;
+    }
+
+    /**
+     * Get the table's charset.
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:24]:
```suggestion
     * Get the table's charaater set.
```

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            final Identifier tableName) {
+        return new Builder(writer, parentSchema, tableName);
+    }
+
+    /**
+     * Builder for {@link MySqlTable}s.
+     */
+    public static class Builder extends Table.Builder {
+
+        private String charset;
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:50]:
```suggestion
    public static class Builder extends Table.Builder {
        private String charset;
```

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+        private Builder(final DatabaseObjectWriter writer, final Schema parentSchema, final Identifier tableName) {
+            super(writer, parentSchema, tableName);
+        }
+
+        @Override
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:56]: Why override the base function if it is unmodified? For the `this` to be the parent type?

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        public Builder column(final String columnName, final String columnType) {
+            super.column(columnName, columnType);
+            return this;
+        }
+
+        /**
+         * Set a custom charset for the new table. Defaults to UTF-8.
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:63]:
```suggestion
         * Set a custom character set for the new table. Defaults to UTF-8.
         * <p>
         * This character set is then used for the whole table down to the columns.
         * Additionally the standard collation rules for this dataset are applied.
         * </p>
```

File: src/test/java/com/exasol/dbbuilder/dialects/mysql/MySqlTableTest.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.exasol.dbbuilder.dialects.DatabaseObjectWriter;
+import com.exasol.dbbuilder.dialects.Schema;
+
+@ExtendWith(MockitoExtension.class)
+class MySqlTableTest {
+
+    @Mock
```
Comment (redcatbear) [inline src/test/java/com/exasol/dbbuilder/dialects/mysql/MySqlTableTest.java:17]:
```suggestion
class MySqlTableTest {
    @Mock
```

**Review by redcatbear — CHANGES_REQUESTED** [review]
Couple small suggestions.

File: src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java (hunk 6 above)
Comment (kaklakariada) [reply src/main/java/com/exasol/dbbuilder/dialects/mysql/MySqlTable.java:54]: Yes, exactly. I added a comment to clarify this.

**Review by redcatbear — APPROVED** [review]

**Pull request merged**